cypress/component/highlight_text.spec.js

```diff
@@ -0,0 +1,174 @@
+/* eslint-disable no-undef */
+import HighlightText from '../../src/components/HighlightText';
+import { mount } from 'cypress/react';
+import React from 'react';
+
+describe('HighlightText - Tests', function() {
+  it('Renders text with no highlighting if no matches', function() {
+    const lorem = 'Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod tempor incididunt ut labore et dolore magna aliqua. Ut enim ad minim veniam, quis nostrud exercitation ullamco laboris nisi ut aliquip ex ea commodo consequat. Duis aute irure dolor in reprehenderit in voluptate velit esse cillum dolore eu fugiat nulla pariatur. Excepteur sint occaecat cupidatat non proident, sunt in culpa qui officia deserunt mollit anim id est laborum.';
+
+    mount(
+      <HighlightText
+        highlight={[
+          {
+            bgColor: 'black',
+            textColor: 'black',
+            words: ['unrelated', 'unused', 'words'],
+          }
+        ]}
+        text={lorem}
+      />
+    );
+    // first highlight chunk is entire text
+    cy.get('[data-cy="highlight-0"]').contains(lorem);
+    // no further chunks
+    cy.get('[data-cy="highlight-1"]').should('not.exist');
+  });
+
+  it('Highlights with the correct color, irrelevant of casing.', function() {
+    const text = (
+      'The quick brown fox jumps over the lazy log.'
```

Comment:
Suggestion - add a variety of casing here, something like: 
```suggestion
      'ThE quick BroWn fox jUMpS over tHe laZy log.'
```